### PR TITLE
Polish cache hit logic

### PR DESCRIPTION
**Pull request by Bambooin**
## Pull request

- fix(ci): extract essential elements for cache hit
- chore(ci): use SHA-1 algorithm for cache key is enough

#### Issue tracker
Fixes will automatically close the related issue

Fixes #

#### Feature
Describe feature of pull request

#### Code of conduct
- [x] [CONTRIBUTING](CONTRIBUTING.md)

#### Gradle task
Tasks passed on every commit
- [x] `./gradlew spotlessCheck`
- [x] `./gradlew assembleDebug`

#### Manual test
- [x] Done

#### Code Review
1. No wildcards import
2. Manual build and test pass
3. GitHub action ci pass
4. At least one contributor reviews and votes
5. Can be merged clean without conflicts
6. PR will be merged by rebase upstream base

#### Daily build
Login to fetch artifact

https://github.com/osfans/trime/actions

### CODE REVIEW

File: script/cache-hash.sh [hunk 1]
```diff
@@ -1,10 +1,21 @@
 #!/usr/bin/env bash
 
-JNI_FILES="magic.txt app/build.gradle app/src/main/jni/cmake/* app/src/main/jni/librime_jni/* app/src/main/jni/CMakeLists.txt"
+DIGEST_ALGORITHM=sha1sum
```
Comment (iovxw) [inline script/cache-hash.sh]: https://en.wikipedia.org/wiki/SHA-1 SHA-1 is insecure, SHA-256 is the [default algorithm of Github Action](https://docs.github.com/en/actions/learn-github-actions/expressions#hashfiles)

**Review by iovxw — CHANGES_REQUESTED** [review]

**Comment by Bambooin** [comment]
Yes, but I think is ok for cache key

I have saw in `man sha1sum`:
`Do not use the SHA-1 algorithm for security related purposes.`

**Comment by iovxw** [comment]
> Yes, but I think is ok for cache key

See the [version rollback attack](https://en.wikipedia.org/wiki/Downgrade_attack). If the attacker sends an evil PR (e.g. with some changes in submodules that make a hash collision,  it's hard to detect), and we merged it, the next release may have an old vulnerable JNI binary

**Comment by Bambooin** [comment]
I already swith to SHA-2 by your advice and code review

**Review by iovxw — APPROVED** [review]

**Pull request merged**